Sources/Support/StoreMessagesHelper.swift

```diff
@@ -0,0 +1,191 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreMessagesHelper.swift
+//
+//  Created by Antonio Rico Diez on 27/9/23.
+
+import StoreKit
+
+class StoreMessagesHelper {
+
+    private let systemInfo: SystemInfo
+    private let showStoreMessagesAutomatically: Bool
+    private let storeMessagesProvider: StoreMessagesProvider
+
+    private var deferredMessages: [StoreMessage] = []
+
+    init(systemInfo: SystemInfo,
+         showStoreMessagesAutomatically: Bool) {
+        self.systemInfo = systemInfo
+        self.showStoreMessagesAutomatically = showStoreMessagesAutomatically
+        self.storeMessagesProvider = StoreMessagesProviderWrapper()
+    }
+
+    @available(iOS 16.0, *)
+    init(systemInfo: SystemInfo,
+         showStoreMessagesAutomatically: Bool,
+         storeMessagesProvider: StoreMessagesProvider = StoreMessagesProviderWrapper()) {
+        self.systemInfo = systemInfo
+        self.showStoreMessagesAutomatically = showStoreMessagesAutomatically
+        self.storeMessagesProvider = storeMessagesProvider
+    }
+
+    #if os(iOS)
+
+    @available(iOS 16.4, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    func deferMessagesIfNeeded() async throws {
+        guard !self.showStoreMessagesAutomatically else {
+            return
+        }
+        Task(priority: .background) { [weak self] in
```

Comment:
not sure this should have low priority, since if this takes long to execute we won't start listening in time, right?

Reply:
Hmm that's a good point... Will increase it to medium priority 👍